src/librustdoc/test.rs

```diff
@@ -370,7 +371,7 @@ pub fn make_test(s: &str,
 
     // Uses libsyntax to parse the doctest and find if there's a main fn and the extern
     // crate already is included.
-    let (already_has_main, already_has_extern_crate, found_macro) = crate::syntax::with_globals(|| {
+    let (already_has_main, already_has_extern_crate, found_macro) = with_default_globals(|| {
```

Comment:
Rustdoc has the edition option as well, it should probably be passed to here instead of the default.